### ISSUE TITLE
`@ts-nocheck` and `@ts-ignore` should also disable suggestions

### ISSUE THREAD

**Issue by gomain** [microsoft/TypeScript#47315]
# Suggestion

## 🔍 Search Terms

@ts-ignore tsserver disable

## ✅ Viability Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, new syntax sugar for JS, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

## ⭐ Suggestion

The `// @ts-nocheck` and `// @ts-ignore` directive should, also, disable tsserver providing suggestions

## 📃 Motivating Example

When writing code ,it's nice that we get _real time_ compile & linting errors & warnings (i.e. suggestion). TSSERVER, the language server for typescript, dose the heavy-lifting for your favorite IDE and fills your screen with all those wiggly lines you hate (& love).

Sometime, though, the language server can get it wrong. You are using a feature that's too new. You depend on a library unknown to the server. Or that you are writing buggy code intentionally. To our rescue, we can use _directives_ to tell typescript (and its language server) that "it's ok, I know, roll with it". Directives such as
```ts
// @ts-nocheck: ignore this file
// @ts-ignore: ignore next line
```
are very handy.

Unfortunately, these directives didn't work with _suggestions_. Things like
```
'foo' is declared but its value is never read. [6133]
File is a CommonJS module; it may be converted to an ES module. [80001]
```
Still populated your IDE, and you jump through them like hoops trying to reach the _error_ you want to fix. They are not _ignored_ by placing the directives.

Released with typescript X.X, these directives can now be used to switch off suggestions as well.

## 💻 Use Cases
![ts-ignore](https://user-images.githubusercontent.com/4217249/148164214-cfa2dcea-96f2-437e-9d83-a46c1de03aba.png)
This is a capture from Emacs' lsp-mode. For conventional reasons, we still want to continue using the commonjs style `require` and `module.exports` in js projects.

**Comment by RyanCavanaugh** [comment]
> Still populated your IDE, and you jump through them like hoops trying to reach the error you want to fix. 

Not to render the original suggestion moot, but this seems like a bug in the editor. VS Code, for example, will not navigate to a Suggestion diagnostic when you invoke the "Go to next error" command.

**Comment by gomain** [comment]
> this seems like a bug in the editor. VS Code, for example, will not navigate to a Suggestion diagnostic when you invoke the "Go to next error" command.

Yes, in Emacs (Flycheck) we can set a minimum error level too. But that is opting out of all suggestions. Some suggestions are useful and (by choice) treated as errors - goto & fix.

We should be able to selectively switch off suggestions, like we can do with errors.

Even better, a way to switch of specific suggestions for the entire project (i.e. via tsconfig) either by code or by class. But with directives is a good start.

**Comment by kshetline** [comment]
I'm adding my voice as one more person who wants @ts-ignore to work with suggestions just as it does with errors.

**Comment by mortie** [comment]
Hi, this is actually pretty important. Suggestions are useless if I can't either fix them or suppress them, and there are situations where fixing them is impossible. With no ability to suppress individual suggestions, the next best alternative is to disable suggestions altogether, which isn't great.

**Comment by RyanCavanaugh** [comment]
Duplicate of #46686
